src/views/PageKYCStatus.vue

```diff
@@ -0,0 +1,269 @@
+
+            <!-- Status Column -->
+            <b-table-column
+                field="KYC Status"
+                filter
+                label="KYC Status"
+                width="150px"
+                sortable
+                searchable
+            >
+                <template #searchable="props">
+                    <AtomSelectInput
+                        :size="'is-small'"
+                        :list="kycStatusList"
+                        class="column-width"
+                    >
+                    </AtomSelectInput>
+                </template>
+                <template v-slot="props">
+                    <SelectInput
+                        :key="props.row.ID"
+                        :defaultValue="getKYCStatusLabel(props.row.KYCStatus)"
+                        :list="kycStatusList.map((kycStatus) => kycStatus.name)"
+                        @change="onStatusUpdate(props.row, $event.target.value)"
+                        name="updateKYCStatus"
+                    />
+                </template>
+            </b-table-column>
+        </b-table>
+    </div>
+</template>
+
+<script>
+import { getIdBasedOnLable, getKYCStatusLabel } from '@/constant/enums';
+import { mapState, mapActions } from 'vuex';
+import AtomSelectInput from '../components/atoms/AtomSelectInput.vue';
+import LoaderModal from '../components/extras/LoaderModal.vue';
+import SelectInput from '@/components/global/SelectInput.vue';
+
+export default {
+    name: 'KYCStatusPage',
+    components: {
+        AtomSelectInput,
+        LoaderModal,
+        SelectInput,
+    },
+
+    data() {
+        return {
+            kycStatusList: [
+                { id: 0, name: 'Not Set' },
+                { id: 1, name: 'Pending' },
+                { id: 2, name: 'ID Verified' },
+                { id: 3, name: 'Ownership Verified' },
+                { id: 4, name: 'Denied' },
+                { id: 5, name: 'Not Verified' },
+            ],
+        };
+    },
+    computed: {
+        ...mapState('kycStatusPortal', [
+            'isLoading',
+            'hasError',
+            'errorMessage',
+            'users',
+        ]),
+    },
+    mounted() {
+        this.fetchUsers();
+    },
+    methods: {
+        ...mapActions('kycStatusPortal', ['fetchUsers', 'updateStatus']),
+
+        // Get label for status based on the enum value
+        getKYCStatusLabel(spotRequestStatus) {
+            return getKYCStatusLabel(spotRequestStatus);
+        },
+
+        alertError(msg) {
+            this.$buefy.dialog.alert({
+                title: 'Error',
+                message: msg,
+                type: 'is-danger',
+                hasIcon: true,
+                icon: 'alert-circle',
+                ariaRole: 'alertdialog',
+                ariaModal: true,
+            });
+        },
+
+        async onStatusUpdate(row, newStatus) {
+            newStatus = getIdBasedOnLable(this.kycStatusList, newStatus);
+            if (newStatus) {
```

Comment:
what if ID returns zero. Zero is a valid enum value, but this if condition would fail.